test/call-test.js

```diff
@@ -852,7 +853,10 @@ describe("sinonSpy.call", function () {
         });
 
         // https://github.com/sinonjs/sinon/issues/1066
+        /* eslint-disable consistent-return */
         it("does not throw when the call stack is empty", function (done) {
+            if (!supportPromise) { return this.skip(); }
```

Comment:
You don't need to return here. To my knowledge, `this.skip()` throws a special error (`SkipError`?) that mocha is expecting so execution should stop immediately. Then you can remove the eslint toggle.

MINOR: You can inline the conditional since it's only used/checked once.